tests/unit_tests/test_config.py

```diff
@@ -146,4 +150,91 @@ def test_auth_from_env_throws_when_not_available():
     with pytest.raises(KeyError):
         BasicAuthentication(username="${BAZ}", password="baz")
     with pytest.raises(KeyError):
-        BasicAuthentication(username="${baz}", password="baz")
+        BasicAuthentication(username="${baz}", passcode="baz")
+
+
+def is_subset(subset: Mapping[str, Any], superset: Mapping[str, Any]) -> bool:
+    """
+    Recursively check if 'subset' is contained within 'superset',
+    skipping nullable (None) fields in the superset.
+    """
+    for key, value in subset.items():
+        if key not in superset:
+            return False
+        superset_value = superset[key]
+
+        # If both values are dictionaries, recurse
+        if isinstance(value, dict) and isinstance(superset_value, dict):
+            if not is_subset(value, superset_value):
+                return False
+        # Check equality for non-dict values, ignoring None in superset
+        elif superset_value is not None and value != superset_value:
+            return False
+    return True
+
+
+# Parameterize the fixture to accept different config examples
+@pytest.fixture
+def temp_yaml_config_file(request):
+    # Use the provided config data from test parameters
+    config_data = request.param
+
+    # Create a temporary YAML file with the configuration
+    with tempfile.NamedTemporaryFile(
+        suffix=".yaml", mode="w", delete=False
+    ) as temp_yaml_file:
+        yaml.dump(config_data, temp_yaml_file)
+        temp_yaml_file_path = temp_yaml_file.name
+
+    # Provide the path and the config data
+    yield Path(temp_yaml_file_path), config_data
+
+    # Cleanup after test execution
+    os.remove(temp_yaml_file_path)
```

Comment:
Should: This seems like a lot of extra boilerplate compared to just writing the example config below into a YAML file.

Reply:
Perhaps, the main motivation is that you may as well have the thing the user will write as an input to your test data

Reply:
outdated